pyInterface/module/_integrals.py

```diff
@@ -30,7 +30,14 @@ def calcIntegrals(integralFileName, eventAndAmpFileDict, multiBin, weightFileNam
 				pyRootPwa.utils.printErr("could not read metadata from amplitude file '" + ampFileName + "'. Aborting...")
 				return False
 			ampMetas.append(ampMeta)
-			integralMetaData.addKeyFileContent(ampMeta.keyfileContent())
+
+			if len(integrals) == 1: # first eventFieldID of the current bin
+				if not integralMetaData.addKeyFileContent(ampMeta.keyfileContent()):
+					pyRootPwa.utils.printWarn("could not add the keyfile content.")
+			else:
+				if not integralMetaData.hasKeyFileContent(ampMeta.keyfileContent()):
+					pyRootPwa.utils.printWarn("keyfile content of additional eventFiledID missing in first eventFieldID.")
```

Comment:
Again, I think that this should be a fatal error, so probably `printErr` instead of `printWarn` should be used, also here `return false` should be added.